src/resources/parser/glb-parser.js

```diff
@@ -1768,6 +1780,17 @@ const createMaterials = function (gltf, textures, options, flipV) {
     });
 };
 
+const createVariants = function (gltf) {
+    if (!gltf.hasOwnProperty("extensions") || !gltf.extensions.hasOwnProperty("KHR_materials_variants"))
+        return null;
+
+    const variants = {};
+    for (let i = 0; i < gltf.extensions.KHR_materials_variants.variants.length; i++) {
```

Comment:
store `gltf.extensions.KHR_materials_variants.variants` in some local variable to avoid all these lookups.